spec/spec_helper.rb

```diff
@@ -65,3 +65,10 @@ def expect_post_with_unprocessable_entity(url, response, data: {})
   exception = RestfulResource::HttpClient::UnprocessableEntity.new(request, rest_client_response)
   expect(@mock_http).to receive(:post).with(url, data: data, headers: {}, open_timeout: nil, timeout: nil).and_raise(exception)
 end
+
+def expect_delete_with_unprocessable_entity(url, response)
```

Comment:
Pedantic points, but....

1. It would be more consistent to write this method on line 47, not at the bottom.

2. These methods are very repetitive!! Do you think it would be better to just define something like:

```
def expect_success(url, response, method:, data: {})
  # ...
end

def expect_unprocessible_entity(url, response, method:, data: {})
  # ...
end
```
?

Reply:
1. Actually, I defined the method at the end for consistency with the entire file.
   1a. Is it possible to be "more consistent"? :trollface: 

2. I see the benefit, but then you need to add some logic inside the methods because not all the _verbs_ receive `data` or any other exception.
There are many places in this gem that we could refactor but I don't want to spend more time on that now if it's not really needed